lib/spack/spack/cmd/__init__.py

```diff
@@ -177,6 +178,33 @@ def parse_specs(
     if not concretize:
         return specs
 
+    # Special case for concretizing a single spec
+    if len(specs) == 1:
+        return [specs[0].concretized()]
+
+    # Special case if every spec has an abstract spec
+    if all(spec.concrete or spec.abstract_hash for spec in specs):
+        ret = [s.concretized() for s in specs]
+
+        unify = spack.config.get("concretizer:unify", False)
+        if unify is True:  # True, "when_possible", False are possible values
+            specs_per_name = Counter(
+                spec.name
+                for spec in traverse.traverse_nodes(
+                    ret, deptype=("link", "run"), key=traverse.by_dag_hash
+                )
+                # TODO: make this less special -- client code shouldn't need to know this.
+                if spec.name != "gcc-runtime"  # gcc runtime is special and allowed multiples
```

Comment:
Should be `spec.name not in pack.repo.PATH.packages_with_tags("runtime")`

Reply:
hm why not:

```python
"runtime" not in spec.package.tags
```

Reply:
apparently that isn't always present. Oh well.